SourceryRuntime/Sources/TypeName.swift

```diff
@@ -70,10 +70,16 @@ public protocol Typed {
         } else {
             name = name.bracketsBalancing()
             name = name.trimmingPrefix("inout ").trimmingCharacters(in: .whitespacesAndNewlines)
-            let isImplicitlyUnwrappedOptional = name.hasSuffix("!") || name.hasPrefix("ImplicitlyUnwrappedOptional<")
-            let isOptional = name.hasSuffix("?") || name.hasPrefix("Optional<") || isImplicitlyUnwrappedOptional
-            self.isImplicitlyUnwrappedOptional = isImplicitlyUnwrappedOptional
-            self.isOptional = isOptional
+
+            if name.isValidClosureName() {
+                let isImplicitlyUnwrappedOptional = name.hasPrefix("ImplicitlyUnwrappedOptional<") && name.hasSuffix(">")
+                self.isImplicitlyUnwrappedOptional = isImplicitlyUnwrappedOptional
+                self.isOptional = (name.hasPrefix("Optional<")  && name.hasSuffix(">")) || isImplicitlyUnwrappedOptional
```

Comment:
I think we should change the doc for this property since it now will return `true` for `!` types, not just `?` types, when before they were mutually exclusive.

Reply:
it worked same way before, you can see that in diff it had the `|| isImplicitlyUnwrappedOptional`